assets/elm/main.elm

```diff
@@ -140,6 +144,34 @@ namesDecoder =
     Json.Decode.field "names" (Json.Decode.list Json.Decode.string)
 
 
+type alias Player =
+    { player_id : Int
+    , name : String
+    , isActive : Bool
+    , seat : Int
+    }
+
+
+playerDecoder : Json.Decode.Decoder Player
+playerDecoder =
+    Json.Decode.map4 Player
+        (Json.Decode.field "id" Json.Decode.int)
+        (Json.Decode.field "name" Json.Decode.string)
+        (Json.Decode.field "is_active" Json.Decode.bool)
+        (Json.Decode.field "seat" Json.Decode.int)
+
+
+type alias GameState =
+    { players : List Player
+    }
+
+
+gameDecoder : Json.Decode.Decoder GameState
+gameDecoder =
+    Json.Decode.map GameState
+        (Json.Decode.field "players" (Json.Decode.list playerDecoder))
```

Comment:
This decoder stuff is pretty cool! It's all very modular and makes sense to read

Reply:
Ya, once I saw how it worked it really isn't so bad at all. As intuitive as it seems like it can be.